### ISSUE TITLE
[Feature]: Warn user about low `niterations`

### ISSUE THREAD

**Issue by MilesCranmer** [MilesCranmer/PySR#603]
### Feature Request

I see a lot of people using the default niterations (purposefully small) for “production runs” and not getting as good results. Either the default `niterations` should be higher, or, unless set explicitly, should raise a warning and point the user to the tuning page.

**Comment by mvanhorn** [comment]
Implemented in https://github.com/MilesCranmer/PySR/pull/1192.

**Comment by MilesCranmer** [comment]
I don't think it should literally be to warn if the defaults are left as-is, as I think that would be pretty annoying for users. The right way is probably to have a measure of problem complexity and warn at that time.

But niterations is probably not the right knob anyways, I think the upcoming `effort` feature is: https://github.com/astroautomata/SymbolicRegression.jl/pull/585

**Comment by mvanhorn** [comment]
Got it -- thanks for the steer. The "warn if defaults are unchanged" shape is wrong; the warning needs to fire on real signal (problem complexity, or once `effort` lands as the canonical knob). I'll back off the warning-pattern PRs (#1193, #1194 closed for the same reason) and watch SymbolicRegression.jl#585.
